rust/src/array.rs

```diff
@@ -86,13 +90,25 @@ impl Array {
         }
     }
 
+    /// Get a reference to the array data
     pub fn data(&self) -> &ArrayData {
         &self.data
     }
 
+    /// number of elements in the array
     pub fn len(&self) -> usize {
         self.len as usize
     }
+
+    /// number of null elements in the array
+    pub fn null_count(&self) -> usize {
+        self.null_count as usize
+    }
+
+    /// If null_count is greater than zero then the validity_bitmap will be Some(Bitmap)
+    pub fn validity_bitmap(&self) -> &Option<Bitmap> {
```

Comment:
I don't think it's a good idea to expose validity bitmap directly to user. I think we should just add methods to test whether a position is set.

Reply:
We have seen some cases in C++ that it is beneficial in a lot of cases to work directly on the bitmap as the compiler can drastically speedup access to it using SIMD instructions.